buildconfig/CMake/MSVCSetup.cmake

```diff
@@ -179,8 +179,10 @@ set(MANTIDPYTHON_PREAMBLE
 # Launch script
 # Developer verison
 set(MANTIDLAUNCH_PREAMBLE
-    "$QT_PLUGIN_PATH=\"${THIRD_PARTY_DIR}\\lib\\qt5\\plugins\"\n$PYTHONHOME=\"${THIRD_PARTY_DIR}\\lib\\python3.8\"\n$ERROR_REPORTER_DIR=\"${PROJECT_SOURCE_DIR}\\scripts\\ErrorReporter\"\n
-    $LAUNCH_SCRIPT=\"workbench-script.pyw\""
+    "$QT_PLUGIN_PATH=\"${THIRD_PARTY_DIR}\\lib\\qt5\\plugins\"
+$PYTHONHOME=\"${THIRD_PARTY_DIR}\\lib\\python3.8\"
+$ERROR_REPORTER_DIR=\"${PROJECT_SOURCE_DIR}\\scripts\\ErrorReporter\"
+$LAUNCH_SCRIPT=\"$scriptRootDirectory\\workbench-script.pyw\""
 )
 configure_file ( ${PACKAGING_DIR}/launch_workbench.ps1.in
     ${PROJECT_BINARY_DIR}/launch_workbench.ps1.in @ONLY )
```

Comment:
Can you remind me of the intention of this one? It's going to the build directory but I'm not sure that will work? I think it needs to go to `${MSVC_BIN_DIR}` and be renamed `launch_workbench.ps1`?

Reply:
So essentially this generates a .in folder in the /bin directory and then using a `file` call generates the .ps script in the correct MSVC directory, see https://github.com/mantidproject/mantid/blob/339e7f155754917c098c23b6f2c1af645f9a2da4/buildconfig/CMake/MSVCSetup.cmake#L190.

It’s a bit convoluted, but I wasn't sure if it could be done in one stage, as this is exactly how the mantidpython.bat file is generated as well, so I maybe wrongly followed suit. https://github.com/mantidproject/mantid/blob/339e7f155754917c098c23b6f2c1af645f9a2da4/buildconfig/CMake/MSVCSetup.cmake#L165

Reply:
Ah yes thank you for reminding me. I don't think `configure_file` works with generator expressions https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html so you have this two step process to get it in the appropriate bin directory. Thanks for looking over that.